### PR TITLE
feat: add node graceful shutdown kubelet config

### PR DESCRIPTION
**Pull request by weichch**
**What type of PR is this?**

/kind feature
/kind api-change

**What this PR does / why we need it**:

This PR adds node graceful shutdown Kubelet config items and exposes those for client use, so that we can allow customers to enable this feature on their nodes.

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

**Comment by cameronmeissner** [comment]
you'll want to recreate this PR without using a fork so required checks can run - I'm gonna add a note about this to the readme so folks don't continue to make the same mistake

File: pkg/agent/datamodel/types.go [hunk 1]
```diff
@@ -2033,6 +2035,22 @@ type AKSKubeletConfiguration struct {
 	Default: []
 	+optional. */
 	AllowedUnsafeSysctls []string `json:"allowedUnsafeSysctls,omitempty"`
+	// shutdownGracePeriod specifies the total duration that the node should delay the
+	// shutdown and total grace period for pod termination during a node shutdown.
+	// Default: "0s"
+	// +featureGate=GracefulNodeShutdown
```
Comment (cameronmeissner) [inline pkg/agent/datamodel/types.go:2041]: we'll need to make sure that RP is setteing the `GracefulNodeShutdown` feature gate

Comment (weichch) [reply pkg/agent/datamodel/types.go:2041]: Feature gate is by default enabled in k8s 1.21+, and we can't edit nodes to have this config. Do we still need to enable the feature gate in this case?

**Comment by weichch** [comment]
I'm closing this PR as per comment. Will recreate it from another branch within the repo.

**Pull request closed without merging**